### DISCUSSION TITLE
Question about live tiles and expiration threshold

### DISCUSSION THREAD

**Discussion by pskpg** [pelicanmapping/osgearth#1944] [Support Q&A] [answered]
Hi, experts!

I'm using osgearth 2.10 and running osgearth_viewer with the earth file below:

![image](https://user-images.githubusercontent.com/20101178/151665301-5abc61a7-722a-4598-9cdd-d3ee9ccddc1a.png)

I found even using cache system, the rendering speed is still relatively low.  For example, when I click A1 viewpoint, I'll have to wait perhaps 3 seconds before the image is ok. 

What' more, when I've the viewpoint A1 loaded before, and revisit it from A2 (or A3, A4, etc. ), the A1 viewpoint performs the loading and rendering procedure again, which takes me another 3 second!

By searching the forum, I learned to increase the _expirationThreshold to keep the loaded tiles alive. And this does enhance my using experiance. But another problem comes: **When I set _expirationThreshold  to 1000,000, the application will become no response after I've visited A1, A2, to about A9 or A10. Through debug tracing, I found the liveTiles' size is about 20,000 when the problem occurs.**

Firstly, I guessed the max capability of liveTiles vector is about 20,000. But, debug tracing shows the number is extremely large (I havn't wrote it down). Then I doubted the DDR memory limites the liveTiles' size. But when I run the application with another 16GB memory computer, the problem happens again.

Could you help me to solve this problem? What should I do to increase the liveTiles' size?

By the way, could you tell me how to config the _expirationThreshold parameter through earth file, please?

Sincere thanks!

**Comment by gwaldron** [comment, marked as answer]
It sounds like your loading speed is too slow for you, not rendering speed. The most likely culprit is your elevation data - make sure it is properly prepared for fast reading. There are some tips here: 
https://docs.osgearth.org/en/latest/data.html#optimizing-a-geotiff
